src/nix/profile.cc

```diff
@@ -451,15 +476,26 @@ class MixProfileElementMatchers : virtual Args
     {
         std::vector<Matcher> res;
 
+        auto anyIndexMatchers = false;
+
         for (auto & s : _matchers) {
-            if (auto n = string2Int<size_t>(s))
+            if (auto n = string2Int<size_t>(s)) {
                 res.push_back(*n);
+                anyIndexMatchers = true;
+            }
             else if (store->isStorePath(s))
                 res.push_back(s);
             else
                 res.push_back(RegexPattern{s,std::regex(s, std::regex::extended | std::regex::icase)});
         }
 
+        if (anyIndexMatchers) {
+            warn("Indices are deprecated and will be removed in a future version!\n"
+                 "         Refer to packages by their `Name` as printed by `nix profile list`.\n"
+                 "         If you want to vote against this change, please leave a comment here:\n"
+                 "         https://github.com/NixOS/nix/issues/9171");
```

Comment:
Is this common practice in nix output? I don't think I've seen issues mentioned like this before, at least.

I understand wanting to give users agency as much as possible, but I fear this may give this issue a bit too much attention; Perhaps it's best to just let people annoyed enough by it find the issue themselves through the commit history?

Reply:
It is not. I wanted to raise this as an idea to allow users to participate, and gain UX feedback without having to collect metrics. Personally, I would appreciate a message like this, it makes me feel like I can be part of a discussion and decisions aren't forced on me, but I'm of course not the average user.

Maybe we can try it out and see how people react to it?

Reply:
I think that's reasonable, but we shouldn't use the word "vote". Maybe "If this would severely impact your setup, please leave a comment explaining your use case..." or something along these lines.

Reply:
Maybe just:

> See https://github.com/NixOS/nix/issues/9171 for more information

I really like that an URL is added :+1:, instead of guessing/searching the right place to discuss such changes further.

Reply:
Yeah that's a good message. I changed it to that.